clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java

```diff
@@ -201,6 +192,27 @@ public void onBecomeOfflineFromError(Message message, NotificationContext contex
     logger.warn("unexpected state transition from ERROR to OFFLINE");
   }
 
+  private void updateLagMonitor(String resourceName, BiConsumer<Version, Integer> lagMonFunction) {
+    try {
+      String storeName = Version.parseStoreFromKafkaTopicName(resourceName);
+      int storeVersion = Version.parseVersionFromKafkaTopicName(resourceName);
+      Pair<Store, Version> res = getStoreRepo().waitVersion(storeName, storeVersion, Duration.ofMillis(500));
```

Comment:
500 millis is not going to be enough. The waitVersion retries this call every 1 second. Any concern in making this timeout to 30 seconds - and probably configurable through some config?

Reply:
Good catch. Let me bump it to higher value.

Reply:
done